payment-manager/src/main/java/com/qpay/paymentmanager/service/impl/WalletServiceImpl.java

```diff
@@ -22,6 +23,14 @@ public WalletEntity getWalletById(final long id) {
         return walletRepository.findById(id).orElseThrow(NoWalletFoundException::new);
     }
 
+    public WalletEntity getWalletByUser(final long userId, final UserType userType) {
+        final WalletEntity walletEntity = walletRepository.findWalletEntityByUserIdAndUserType(userId, userType);
+        if (walletEntity == null) {
+            throw new NoWalletFoundException();
+        }
```

Comment:
Use the same approach of checking null as you did in another method to get by id

Reply:
![image](https://github.com/Hoocki/qpay/assets/75415127/ee209f90-6196-4d9e-9634-0f8baafe2b62)

i cant

Reply:
make a repository method return type `optional`